src/StaticValidator.php

```diff
@@ -47,14 +47,22 @@ public static function getPluginManager()
     /**
      * @param  mixed    $value
      * @param  string   $classBaseName
-     * @param  array    $args          OPTIONAL
+     * @param  array    $options OPTIONAL associative array of options to pass as
+     *     the sole argument to the validator constructor.
      * @return bool
+     * @throws Exception\InvalidArgumentException for an invalid $options argument.
      */
-    public static function execute($value, $classBaseName, array $args = [])
+    public static function execute($value, $classBaseName, array $options = [])
     {
+        if ($options && array_values($options) === $options) {
```

Comment:
The above is a standard, performant, robust solution that is less likely to obtain false positives. (It is the solution used in `Zend\Stdlib\ArrayUtils::isList()`.)